src/orion/core/worker/trial.py

```diff
@@ -470,9 +470,11 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_value, traceback):
         try:
-            if exc_type is None:
-                self._cm_experiment.release(self._cm_trial)
-            else:
+            if exc_type is KeyboardInterrupt:
+                self._cm_experiment.release(self._cm_trial, "interrupted")
+            elif exc_type is not None:
                 self._cm_experiment.release(self._cm_trial, "broken")
+            else:
+                self._cm_experiment.release(self._cm_trial)
         except:
```

Comment:
I think it would be better to only silence runtime errors here. It may be good to log a warning when catching Runtime errors as well. I say this error in particular because it is the only likely error that may occur because of status issues during `__exit__` (https://github.com/Epistimio/orion/blob/develop/src/orion/client/experiment.py#L448).